lib/models/package-info-cache/package-info.js

```diff
@@ -5,6 +5,24 @@ const ErrorList = require('./error-list');
 const Errors = require('./errors');
 const AddonInfo = require('../addon-info');
 
+function lexicographically(a, b) {
+  if (typeof a.name === 'string' && typeof b.name === 'string') {
+    return a.name.localeCompare(b.name);
+  }
+  return 0;
+}
+
+// lastWriteWins
+function pushUnique(array, entry) {
+  const index = array.indexOf(entry);
+
+  if (index > -1) {
+    array.splice(index, 1);
+  }
+
+  array.push(entry);
+}
+
```

Comment:
I would add a comment above explaining what you're doing here. The 'splice out previous entry and push new entry at the end' feels a little weird unless the reader knows that the list will ultimately get sorted. Otherwise I would do

```
if (index > -1) {
  array.splice(index, 1, entry);
} else {
  array.push(entry);
}